tests/functional/test_config_groups.py

```diff
@@ -0,0 +1,152 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=redefined-outer-name, unused-argument, duplicate-code
+
+from typing import List
+
+import allure
+import pytest
+from adcm_client.objects import (
+    ADCMClient,
+    Cluster,
+    Provider,
+    Host,
+)
+from adcm_pytest_plugin import utils
+from adcm_pytest_plugin.utils import get_data_dir
+from coreapi.exceptions import ErrorMessage
+
+from tests.library.errorcodes import GROUP_CONFIG_HOST_ERROR
+
+CLUSTER_BUNDLE_PATH = get_data_dir(__file__, "cluster_simple")
+PROVIDER_BUNDLE_PATH = get_data_dir(__file__, "hostprovider_bundle")
+
+
+@pytest.fixture()
+def cluster(request, sdk_client_fs: ADCMClient) -> Cluster:
+    bundle_path = request.param if hasattr(request, "param") else CLUSTER_BUNDLE_PATH
+    cluster_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return cluster_bundle.cluster_create(name=utils.random_string())
+
+
+@pytest.fixture()
+def provider(request, sdk_client_fs: ADCMClient) -> Provider:
+    bundle_path = request.param if hasattr(request, "param") else PROVIDER_BUNDLE_PATH
+    provider_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return provider_bundle.provider_create(name=utils.random_string())
+
+
+class TestGroupsIntersection:
+
+    HOST_ERROR_MESSAGE = (
+        "host is not available for this object, or host already is a member of another group of this object"
+    )
+    FIRST_COMPONENT_NAME = "first"
+    SECOND_COMPONENT_NAME = "second"
+    FIRST_GROUP = "test_group"
+    SECOND_GROUP = "test_group_2"
+    FIRST_HOST = "test_host_1"
+    SECOND_HOST = "test_host_2"
+
+    def create_two_hosts(self, provider: Provider):
+        with allure.step("Create host for config groups"):
+            test_host_1 = provider.host_create(fqdn=self.FIRST_HOST)
+        with allure.step("Create host for host candidate check"):
+            test_host_2 = provider.host_create(fqdn=self.SECOND_HOST)
+        return test_host_1, test_host_2
+
+    @allure.step('Check error')
+    def assert_that_host_add_is_unavaliable(self, error):
+        GROUP_CONFIG_HOST_ERROR.equal(error)
+        assert self.HOST_ERROR_MESSAGE in error.value.error['desc']
+
+    @allure.step("Check that the only second host is present in candidates on second provider group")
+    def assert_host_candidate_equal_expected(self, group: List[Host], expected_hosts: int):
+        assert len(group) == expected_hosts, "One host should be available in group"
+        assert group[0].fqdn == self.SECOND_HOST, f"Should be available host '{self.SECOND_HOST}'"
+
+    @pytest.fixture()
+    def cluster_with_components(self, cluster, provider):
```

Comment:
The name of the function is `cluster_with_components`, but it returns service and two hosts.
It's confusing.
Please think about renaming and certainly add typing

Reply:
And add docstring to clarify why or when do wee need this exactly fixture

Reply:
it's impossible to create components without service and host

Reply:
That's right that we need service to create components, but it's not clear why function `cluster_with_components` does not return cluster and it's components, but `Tuple[Service, Host, Host]` (and hosts has hardcoded components)

Reply:
changed, docstring has been added